odoo/models.py

```diff
@@ -4635,6 +4635,34 @@ def _search(self, domain, offset=0, limit=None, order=None, access_rights_uid=No
 
         return query
 
+    def _as_query(self, ordered=False):
```

Comment:
Maybe you are right: making it ordered by default is never incorrect and overall safer.  The two cases where you want to avoid the overhead of making it ordered are:
* a call to `_search()` to make a subquery;
* passing a record as a query to `_fetch_query()` for fetching columns.